src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -39,4 +43,56 @@ public List<Station> getStations() {
                 .distinct()
                 .collect(Collectors.toList());
     }
+
+    public void removeStation(Station station) {
+        Optional<Section> upSection = findSectionbyUpStation(station);
+        Optional<Section> downSection = findSectionbyDownSataion(station);
+
+        checkSectionSize();
+        checkSectionInLine(upSection, downSection);
+
+        if (upSection.isPresent() && downSection.isPresent()) {
+            deleteMiddle(upSection.get(), downSection.get());
+            return;
+        }
+        checkDeleteEnd(upSection, downSection);
+    }
+
+    private void checkSectionInLine(Optional<Section> upSection, Optional<Section> downSection) {
+        if (!upSection.isPresent() && !downSection.isPresent()) {
+            throw new IllegalArgumentException(ErrorMessage.ERROR_NO_STATION_LINE);
+        }
+    }
+
+    private void checkSectionSize() {
+        if (sections.size() <= MIN_SECTION_SIZE) {
+            throw new IllegalArgumentException(ErrorMessage.ERROR_NOT_REMOVE_SECTION);
+        }
+    }
+
+    private void checkDeleteEnd(Optional<Section> upSection, Optional<Section> downSection) {
+        upSection.ifPresent(this::removeEndSection);
+        downSection.ifPresent(this::removeEndSection);
+    }
+
+    private void removeEndSection(Section section) {
+        sections.remove(section);
+    }
+
+    private void deleteMiddle(Section upSection, Section downSection) {
+        downSection.refreshWith(upSection);
+        sections.remove(upSection);
```

Comment:
상태를 변경할 시 최대한 한곳에서 할 수 있도록 하는 것도 괜찮을 것 같아요 😄 
```suggestion
        removeEndSection(upSection);
```

Reply:
👍👍 한곳에서 할 수 있도록 변경하는 편이 가독성면에서도 좋을 것 같습니다! 자세히 알려주셔서 감사합니다